lib/samples/show_device_location_history/show_device_location_history_sample.dart

```diff
@@ -26,26 +26,39 @@ class ShowDeviceLocationHistorySample extends StatefulWidget {
   const ShowDeviceLocationHistorySample({super.key});
 
   @override
-  State<ShowDeviceLocationHistorySample> createState() =>
-      _ShowDeviceLocationHistorySampleState();
+  State<ShowDeviceLocationHistorySample> createState() => Configure();
 }
 
-class _ShowDeviceLocationHistorySampleState
-    extends State<ShowDeviceLocationHistorySample> with SampleStateSupport {
+class Configure extends State<ShowDeviceLocationHistorySample>
+    with SampleStateSupport {
+  // Create a controller for the map view.
   final _mapViewController = ArcGISMapView.createController();
-  double _latitude = 0.0;
-  double _longitude = 0.0;
-  double _heading = 0.0;
+  // A location data source to simulate location updates.
   final _locationDataSource = SimulatedLocationDataSource();
+  // Subscriptions to listen for location status changes.
   StreamSubscription? _statusSubscription;
+  // Subscriptions to listen for location changes.
   StreamSubscription? _locationSubscription;
-  ArcGISException? _ldsException;
+  // A GraphicsOverlay to display the location history polyline.
+  final _locationHistoryLineOverlay = GraphicsOverlay();
+  // A GraphicsOverlay to display the location history points.
+  final _locationHistoryPointOverlay = GraphicsOverlay();
+  // A PolylineBuilder to build the location history polyline.
+  final _polylineBuilder = PolylineBuilder.fromSpatialReference(
+    SpatialReference.wgs84,
+  );
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // A flag for toggling location tracking.
+  var _enableTracking = false;
 
   @override
   void dispose() {
     _locationDataSource.stop();
     _locationSubscription?.cancel();
     _statusSubscription?.cancel();
+    _locationHistoryLineOverlay.graphics.clear();
+    _locationHistoryPointOverlay.graphics.clear();
```

Comment:
I don't think there's a need to clear the graphics. They should be cleaned up when the garbage collector is triggered.

Reply:
Here is my thought:

When a stateful widget (sample) is removed from the UI tree (Sample Viewer), it is not immediately destroyed by garbage collection (GC). Since the sample is retained by our Sample Viewer (_sampleWidget), it will persist until we switch to a new one or terminate the Sample Viewer. This results in unnecessary memory usage, so we need to clean it up as much as possible.

Reply:
I wrote this up in Zenhub (issue 2023). We'll need to address it for all samples. There might be a more global solution -- perhaps the SampleListView can dispose of the sample when we navigate back.